app/pages/learn/01_tutorial/04_mastering-the-api/02_invoke/00_methodhandle.md

```diff
@@ -0,0 +1,626 @@
+
+```java
+setterMethodHandle.invoke(example, "new value");
+String result = (String) getterMethodHandle.invoke(example); // result == `new value`
+```
+
+Although it is possible to read and write field values using method handles, it's not common practice. For fields,
+it's more suitable to use [`VarHandle`](javadoc:VarHandle)s instead, which can be created using
+[`findVarHandle`](javadoc:MethodHandles.Lookup.findVarHandle(Class,String,Class)) and
+[`findStaticVarHandle`](javadoc:MethodHandles.Lookup.findStaticVarHandle(Class,String,Class))
+methods.
+
+<a id="arrays">&nbsp;</a>
+## Working with arrays
+The [`MethodHandles`](javadoc:MethodHandles) class contains methods that provide a number of preset method handles.
+These include method handles that allow array manipulations. Creating these method handles doesn't require access checking,
+so the lookup object is not necessary.
+
+Let's create an array of Strings containing 5 elements using [`arrayConstructor`](javadoc:MethodHandles.arrayConstructor(Class)):
+
+```java
+MethodHandle arrayConstructor = MethodHandles.arrayConstructor(String[].class);
+String[] arr = (String[]) arrayConstructor.invoke(5);
+```
+
+To modify a single element, we can use [`arrayElementSetter`](javadoc:MethodHandles.arrayElementSetter(Class)), to which
+we provide the reference to the target array, the index of an element, and the new value:
+
+```java
+MethodHandle elementSetter = MethodHandles.arrayElementSetter(String[].class);
+elementSetter.invoke(arr, 4, "test");
+```
+
+To read the value of a single element, we should use [`arrayElementGetter`](javadoc:MethodHandles.arrayElementGetter(Class))
+method handle, to which we provide the reference to an array and the element index:
+
+```java
+MethodHandle elementGetter = MethodHandles.arrayElementGetter(String[].class);
+String element = (String) elementGetter.invoke(arr, 4); // element == "test"
+```
+
+We could also use the method handle provided by [`arrayLength`](javadoc:MethodHandles.arrayLength(Class)) to get the array
+length as integer:
+
+```java
+MethodHandle arrayLength = MethodHandles.arrayLength(String[].class);
+int length = (int) arrayLength.invoke(arr); // length == 5
+```
+
+<a id="exceptions">&nbsp;</a>
+## Exception handling
+Both `invokeExact` and `invoke` throw [`Throwable`](javadoc:Throwable), so there is no limitation to what an underlying
+method can throw. The method that invokes a method handle must either explicitly throw a `Throwable` or catch it.
+
+There are certain methods in the `MethodHandles` API that can make exception handling easier. Let's take a look at
+several examples.
+
+### `catch` wrapper
+The [`MethodHandles.catchException`](javadoc:MethodHandles.catchException(MethodHandle,Class,MethodHandle)) method can
+wrap a given method handle inside a provided exception handler method handle.
+
+Say, we have a method `problematicMethod` that does some job, and a method `exceptionHandler` that handles a particular
```

Comment:
Personally, I associate job with batch job 😅 ; just a suggestion as the literature  uses the business logic syntagm more often: _a method `problematicMethod` that performs some business logic_?
```suggestion
Say, we have a method `problematicMethod` that performs some business logic, and a method `exceptionHandler` that handles a particular
```